metadata.go

```diff
@@ -0,0 +1,147 @@
+	MetadataOpenCDCVersion = cpluginv1.MetadataOpenCDCVersion
+
+	// MetadataCreatedAt is a Record.Metadata key for the time when the record
+	// was created in the 3rd party system. The expected format is a unix
+	// timestamp in nanoseconds.
+	MetadataCreatedAt = cpluginv1.MetadataCreatedAt
+	// MetadataReadAt is a Record.Metadata key for the time when the record was
+	// read from the 3rd party system. The expected format is a unix timestamp
+	// in nanoseconds.
+	MetadataReadAt = cpluginv1.MetadataReadAt
+
+	// MetadataConduitPluginName is a Record.Metadata key for the name of the
+	// plugin that created this record.
+	MetadataConduitPluginName = cpluginv1.MetadataConduitPluginName
+	// MetadataConduitPluginVersion is a Record.Metadata key for the version of
+	// the plugin that created this record.
+	MetadataConduitPluginVersion = cpluginv1.MetadataConduitPluginVersion
+)
+
+// MetadataUtil provides utility methods for reading and writing metadata.
+type MetadataUtil struct{}
+
+// SetOpenCDCVersion sets the metadata value for key MetadataVersion to the
+// current version of OpenCDC used. This
+func (MetadataUtil) SetOpenCDCVersion(metadata map[string]string) {
+	metadata[MetadataOpenCDCVersion] = cpluginv1.OpenCDCVersion
+}
+
+// GetOpenCDCVersion returns the value for key
+// MetadataOpenCDCVersion. If the value is does not exist or is empty the
+// function returns ErrMetadataFieldNotFound.
+func (u MetadataUtil) GetOpenCDCVersion(metadata map[string]string) (string, error) {
+	return u.getValue(metadata, MetadataOpenCDCVersion)
+}
+
+// GetCreatedAt parses the value for key MetadataCreatedAt as a unix
+// timestamp. If the value does not exist or the value is empty the function
+// returns ErrMetadataFieldNotFound. If the value is not a valid unix timestamp
+// in nanoseconds the function returns an error.
+func (MetadataUtil) GetCreatedAt(metadata map[string]string) (time.Time, error) {
+	raw := metadata[MetadataCreatedAt]
+	if raw == "" {
+		return time.Time{}, fmt.Errorf("failed to get value for %q: %w", MetadataCreatedAt, ErrMetadataFieldNotFound)
+	}
+
+	unixNano, err := strconv.ParseInt(raw, 10, 64)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("failed to parse value for %q: %w", MetadataCreatedAt, err)
+	}
+
+	return time.Unix(0, unixNano), nil
+}
+
+// SetCreatedAt sets the metadata value for key MetadataCreatedAt as a
+// unix timestamp in nanoseconds.
+func (MetadataUtil) SetCreatedAt(metadata map[string]string, createdAt time.Time) {
+	metadata[MetadataCreatedAt] = strconv.FormatInt(createdAt.UnixNano(), 10)
+}
+
+// GetReadAt parses the value for key MetadataReadAt as a unix
+// timestamp. If the value does not exist or the value is empty the function
+// returns ErrMetadataFieldNotFound. If the value is not a valid unix timestamp
+// in nanoseconds the function returns an error.
+func (MetadataUtil) GetReadAt(metadata map[string]string) (time.Time, error) {
+	raw := metadata[MetadataReadAt]
```

Comment:
It looks like the code we have in GetReadAt can be reused in GetCreatedAt.

Reply:
I'll skip this as well, let's think about it when/if we have another metadata field that contains unix nanos. I think DRY is an overrated principle that is abused in cases like this, we can discuss it some time 🙂

Reply:
I do agree that it's an overrated principle. The reason why I think it would be good to apply in cases like these is because the logic for both is the same, and will likely stay the same for these methods (as it's the same type). But not a big thing anyway. 

> we can discuss it some time slightly_smiling_face

I'll be glad!